src/interpreter/eval.cpp

```diff
@@ -153,13 +153,8 @@ void eval_call_function(gc_ptr<Function> callee, int arg_count, Interpreter& e)
 		e.m_stack.frame_at(offset) = kv.second;
 	}
 
-	// this feels really dumb:
-	// we get a value on the stack, then we move it to the return value
-	// slot, then we pop some stuff off the stack, and put it back on the
-	// stack. It is doubly dumb when we eval a seq-expr (because it does a
-	// save-pop sequence)
 	eval(callee->m_def->m_body, e);
-	e.save_return_value(e.m_stack.pop_unsafe());
+	e.m_stack.frame_at(-1) = e.m_stack.pop_unsafe();
```

Comment:
The `frame_at(-1)` could potentially be the new "store return value" and be called with such nomenclature. Shame that we still have something with that name and that we will change all this to bytecode anyway. So don't pay attention to this comment.